ext/mbstring/php_mbregex.c

```diff
@@ -1361,15 +1360,16 @@ _php_mb_regex_ereg_search_exec(INTERNAL_FUNCTION_PARAMETERS, int mode)
 	OnigUChar *str;
 	OnigSyntaxType *syntax;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|ss", &arg_pattern, &arg_pattern_len, &arg_options, &arg_options_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|s!s!", &arg_pattern, &arg_pattern_len, &arg_options, &arg_options_len) == FAILURE) {
 		RETURN_THROWS();
 	}
 
-	option = MBREX(regex_default_options);
-
 	if (arg_options) {
 		option = 0;
 		_php_mb_regex_init_options(arg_options, arg_options_len, &option, &syntax, NULL);
+	} else {
+		option = MBREX(regex_default_options);
+		syntax = MBREX(regex_default_syntax);
```

Comment:
Hm, this code looks buggy (even before your change), because it just uses MBREX(regex_default_syntax) below.

Reply:
I've addressed this separately in https://github.com/php/php-src/commit/3f2f36d5d4c5c3c77703b366a5c7f9c3682727ee now.

Reply:
Thanks!